### ISSUE TITLE
PopcornTime exited on video playing

### ISSUE THREAD

**Issue by vladshut** [popcorntime/popcorntime#956]
Operating System Version: Ubuntu 18.04

Popcorn Time Version: development branch

Download date: 20.01.2019

#### Expected Behaviour

Play video (film, trailer, etc)

#### Actual Behaviour

Exited

#### Steps to reproduce the behaviour

1. Start program with command yarn gulp run
2. Open any film
3. Click watch now or watch trailer
4. Before video starts to play app exited

Console output: Popcorn-Time exited with code 0
Sometimes I also get error in console: [ERROR:adm_helpers.cc(73)] Failed to query stereo recording.

**Comment by justjoeyuk** [comment]
I can confirm that this exact thing is happening to me. I'm using the "development" branch.

**Comment by alecmerdler** [comment]
Spent some time debugging the source, found a workaround.
1. Add a `console.log(url)` to [this line in `streamer.js`](https://github.com/popcorn-official/popcorn-desktop/blob/development/src/app/lib/streamer.js#L306) to get the HTTP URL of the torrent media
2. Make [`waitForBuffer` do nothing by having it immediately `return null`](https://github.com/popcorn-official/popcorn-desktop/blob/development/src/app/lib/streamer.js#L327)
3. `yarn run start`
4. Open DevTools so you can see the `console.log()` output
5. Use PopcornTime normally to "Watch Now", and copy the URL from the `console.log()`
6. Paste URL (looks like `http://127.0.0.1:50928/1`) into your browser and begin watching

No idea why when the NW.js instance tries to hit that URL endpoint, the entire app crashes.

**Comment by andremeireles** [comment]
Build 361 - 0.3.10-0 (17b2f398) is playing videos and trakt.tv is syncing. I am haven't success with new builds or official download for linux (track.tv sync).
[https://ci.popcorntime.sh/view/All/job/Popcorn-Time-Desktop/361/](https://ci.popcorntime.sh/view/All/job/Popcorn-Time-Desktop/361/)

**Comment by AndrewNelis** [comment]
Also seeing this in Fedora 29.

Following on from @alecmerdler, I've narrowed it down further to the video player playing.
If you remove the `autoplay` tag from `templates/player.tpl`, then the video loads but will still crash if you click to play.

**Comment by AndrewNelis** [comment]
The issue is with the version of ffmpeg causes the crash - this ticket is what's happening: https://github.com/iteufel/nwjs-ffmpeg-prebuilt/issues/70

I changed `gulpfile.js`, line 145 to download a more recent working version:
```
const ffmpegVersion = '0.35.3';
const ffmpegurl = 'https://github.com/iteufel/nwjs-ffmpeg-prebuilt/releases/download/' + ffmpegVersion + '/' + ffmpegVersion + '-' + osvar;
```
Remove the `cache/ffmpeg` folder to cause it to re-download the updated version next run.

**Comment by MarkEhr** [comment]
> The issue is with the version of ffmpeg causes the crash - this ticket is what's happening: [iteufel/nwjs-ffmpeg-prebuilt#70](https://github.com/iteufel/nwjs-ffmpeg-prebuilt/issues/70)
> 
> I changed `gulpfile.js`, line 145 to download a more recent working version:
> 
> ```
> const ffmpegVersion = '0.35.3';
> const ffmpegurl = 'https://github.com/iteufel/nwjs-ffmpeg-prebuilt/releases/download/' + ffmpegVersion + '/' + ffmpegVersion + '-' + osvar;
> ```
> 
> Remove the `cache/ffmpeg` folder to cause it to re-download the updated version next run.

This fixed the crash, but now it's soundless when playing mkv videos. Not sure if it's related but it started after doing this fix.
Sound works with other formats though.

**Comment by devkinetic** [comment]
I get the error when using the butter player, or MPV, but I am able to use VLC.

I should note that this was only after implementing @AndrewNelis fix and with `const ffmpegVersion = '0.44.1';` which is the latest version as of the date of my comment.

I'm using OSX catalina, and I also had to get node-gyp up and running properly (again) in order to rebuild.

**Comment by team-pct** [comment]
Fixed with last build
